src/rocket.rs

```diff
@@ -0,0 +1,12 @@
+use rocket::form::{self, FromFormField, ValueField};
+use std::str::FromStr;
+use crate::Decimal;
+
+impl<'v> FromFormField<'v> for Decimal {
+    fn default() -> Option<Self> {
+        Default::default()
+    }
+    fn from_value(field: ValueField<'v>) -> form::Result<'v, Self> {
+        Decimal::from_str(field.value).map_err(|_| form::Error::validation("not a valid number").into())
```

Comment:
Newbie question: do you need to do anything to handle empty strings? Or is that handled by Rocket if you mark the `Decimal` as `Option`?

Reply:
The latter.

If you require a `Decimal` and `from_value` fails (because of an empty or invalid value), the request fails (similary to how other types can fail, such as ints). If you require an `Option<Decimal>`, values that fail `from_value` become a `None`.

Reply:
Oops my mistake. I've implemented `default`, so when `from_value` (if running on lenient mode, which is the default one) fails it is called and returns `Decimal::default()`, which was not intended and could be misleading.

I'm gonna go ahead and remove that